api/envoy/config/route/v3/route_components.proto

```diff
@@ -147,6 +147,13 @@ message VirtualHost {
   // independently (e.g.: values are not inherited).
   RetryPolicy retry_policy = 16;
 
+  // [#not-implemented-hide:]
+  // Specifies the configuration for retry policy extension. Note that setting a route level entry
+  // will take precedence over this config and it'll be treated independently (e.g.: values are not
+  // inherited). :ref:`Retry policy <envoy_api_field_config.route.v3.VirtualHost.retry_policy>` should not be
+  // set if this field is used.
+  google.protobuf.Any retry_policy_typed_config = 19;
```

Comment:
Thanks. The API no makes sense. I would also explain in the `retry_policy` docs that the converse holds. Also, we should add an example retry exension with typed config to the docs. It's OK to take that as an AI for our next PR.

Reply:
> I would also explain in the retry_policy docs that the converse holds.

I planned to add the exclusion comment in the retry_policy field but this field is marked as hidden and I cannot refer the field in the doc. Maybe I should just put the field there without reference?

I planned to add the retry policy extension example in the following PRs.

Reply:
OK, this is fine for now.